settings/R/read.settings.R

```diff
@@ -35,43 +35,46 @@ check.inputs <- function(settings) {
   if (nrow(inputs) > 0) {
     for(i in 1:nrow(inputs)) {
       tag <- inputs$tag[i]
+      tagid <- paste0(tag, ".id")
       hostname <- settings$run$host$name
       allinputs <- allinputs[allinputs != tag]
 
-      # check if tag exists
-      if (is.null(settings$run$inputs[[tag]])) {
-        if (inputs$required[i]) {
-          logger.severe("Missing required input :", tag)
-        } else {
-          logger.info("Missing optional input :", tag)
-        }
-        next
-      }
-      
-      # check if <id> exists
-      if ("id" %in% names(settings$run$inputs[[tag]])) {
-        id <- settings$run$inputs[[tag]][['id']]
+      # check if <tag.id> exists
+      if (!is.null(settings$run$inputs[[tagid]])) {
+        id <- settings$run$inputs[[tagid]]
         file <- dbfile.file("Input", id, dbcon, hostname)
         if (is.na(file)) {
           logger.error("No file found for", tag, " and id", id, "on host", hostname)
         } else {
-          if (is.null(settings$run$inputs[[tag]][['path']])) {
-            settings$run$inputs[[tag]]['path'] <- file
-          } else if (file != settings$run$inputs[[tag]][['path']]) {
+          if (is.null(settings$run$inputs[[tag]])) {
+            settings$run$inputs[[tag]] <- file
+          } else if (file != settings$run$inputs[[tag]]) {
             logger.warn("Input file and id do not match for ", tag)
           }
         }
```

Comment:
I've moved this bit of the code back ... but without a test it is hard to confirm that the desired functionality is there, or to prevent it from being lost in the future.